kore/src/Kore/Step/Axiom/Identifier.hs

```diff
@@ -108,4 +119,73 @@ matchAxiomIdentifier = Recursive.fold matchWorker . Builtin.externalize
             ExistsF exists -> Exists <$> Syntax.existsChild exists
             VariableF _ ->
                 pure Variable
+            EndiannessF endiannessF ->
+                mkAppId $
+                    Endianness.toApplication $ getConst endiannessF
+            SignednessF signednessF ->
+                mkAppId $
+                    Signedness.toApplication $ getConst signednessF
+            InjF inj -> mkAppId $ Inj.toApplication inj
+            InternalListF internalList -> listToId internalList
+            InternalSetF internalSet -> mapToId internalSet
+            InternalMapF internalMap -> setToId internalMap
             _ -> empty
+
+    listToId internalList
+        | Seq.null list = pure $ Application $ symbolToId unitSymbol
+        | Seq.length list == 1 = pure $ Application $ symbolToId elementSymbol
+        | otherwise = pure $ Application $ symbolToId concatSymbol
+      where
+        InternalList{internalListChild = list} = internalList
+        InternalList{internalListUnit = unitSymbol} = internalList
+        InternalList{internalListElement = elementSymbol} = internalList
+        InternalList{internalListConcat = concatSymbol} = internalList
+
+    mapToId internalMap =
+        acToId
+            unitSymbol
+            elementSymbol
+            concatSymbol
+            elementsWithVariables
+            (HashMap.toList concreteElements)
+            opaque
+      where
+        InternalAc{builtinAcChild} = internalMap
+        InternalAc{builtinAcUnit = unitSymbol} = internalMap
+        InternalAc{builtinAcElement = elementSymbol} = internalMap
+        InternalAc{builtinAcConcat = concatSymbol} = internalMap
+
+        normalizedAc = unwrapAc builtinAcChild
+
+        NormalizedAc{elementsWithVariables} = normalizedAc
+        NormalizedAc{concreteElements} = normalizedAc
+        NormalizedAc{opaque} = normalizedAc
+
+    setToId internalSet =
+        acToId
+            unitSymbol
+            elementSymbol
+            concatSymbol
+            elementsWithVariables
+            (HashMap.toList concreteElements)
+            opaque
+      where
+        InternalAc{builtinAcChild} = internalSet
+        InternalAc{builtinAcUnit = unitSymbol} = internalSet
+        InternalAc{builtinAcElement = elementSymbol} = internalSet
+        InternalAc{builtinAcConcat = concatSymbol} = internalSet
+
+        normalizedAc = unwrapAc builtinAcChild
+
+        NormalizedAc{elementsWithVariables} = normalizedAc
+        NormalizedAc{concreteElements} = normalizedAc
+        NormalizedAc{opaque} = normalizedAc
+
+    acToId unitSymbol _ _ [] [] [] = pure $ Application $ symbolToId unitSymbol
+    acToId _ elementSymbol _ [_] [] [] = pure $ Application $ symbolToId elementSymbol
+    acToId _ elementSymbol _ [] [_] [] = pure $ Application $ symbolToId elementSymbol
+    acToId _ _ _ [] [] [opaque] = opaque
+    acToId _ _ concatSymbol _ _ _ = pure $ Application $ symbolToId concatSymbol
```

Comment:
This is obviously complex. It should have unit tests.

Reply:
We are already exercising this piece of code:
```
    , testGroup
        "Map"
        [ test
            "unitMap"
            (Mock.builtinMap [])
            (Application Mock.unitMapId)
        , test
            "elementMap"
            (Mock.builtinMap [(Mock.a, Mock.a)])
            (Application Mock.elementMapId)
        , test
            "concatMap"
            (Mock.builtinMap [(Mock.a, Mock.a), (Mock.b, Mock.b)])
            (Application Mock.concatMapId)
        ]
```
The above is from `Test.Kore.Step.Axiom.Identifier`. It's true that it isn't technically calling `acToId` directly but the functions that it goes through to get there only do some trivial unpacking of records (that's all `mapToId` and `setToId` do).
Do you think we need anything more than this?